### PR TITLE
Zilliqa transaction signing

### LINKED ISSUE #7: Add support for Zilliqa
Name: [Zilliqa](https://zilliqa.com/)
HD Derivation Scheme: BIP44 / 313
Symbol: `ZIL`

Checklist

- [x] Implement functionality in C++. Put it in a subfolder of `src/`.
    - [x] Address (if necessary)
    - [x] Transaction (if necessary)
    - [x] Signer
- [x] Write unit tests. Put them in a subfolder of `tests/`.
- [x] Add relevant constants in `TWCoinType`, `TWP2SHPrefix`, `TWEthereymChainID`, `TWHRP`, etc., as necessary.
- [x] Return correct curve and purpose in `src/Coin.cpp`.
- [x] Implement address validation and derivation in `src/Coin.cpp`.
- [x] Implement coin configuration `src/include/TWCoinTypeConfiguration.cpp`.
- [x] Write interface header in `include/TrustWalletCore` and implement the interface in `src/interface`.
    - [x] Address interface (if necessary).
    - [x] Signing interface.
- [x] Validate generated code in Android an iOS projects. Write integration tests for each.

### PR DESCRIPTION
**Pull request by hewigovens**
Fix #7

### CODE REVIEW

File: src/Zilliqa/AddressChecksum.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/// see https://github.com/Zilliqa/Zilliqa/blob/1c53b792c7ae44f7b77366536a7e2f73a3eade6a/src/libServer/AddressChecksum.h
+std::string Zilliqa::checksumed(const Address& address) {
+    const auto addressString = hex(address.bytes);
+    const auto hash = hex(Hash::sha256(address.bytes));
+
+    uint256_t temp_1 = 1;
+    uint256_t v("0x" + hash);
+
+    std::string string = "0x";
+    for (auto i = 0; i < addressString.size(); i += 1) {
```
Comment (vikmeup) [inline src/Zilliqa/AddressChecksum.cpp]: did you consider addsing support for a new type here: https://github.com/TrustWallet/wallet-core/blob/master/src/Ethereum/AddressChecksum.cpp#L16 ?

File: src/Zilliqa/Address.cpp [hunk 2]
```diff
@@ -29,6 +30,13 @@ Address::Address(const PublicKey& publicKey) {
     std::copy(data.end() - Address::size, data.end(), bytes.begin());
 }
 
+Address::Address(const Data& data) {
+    if (!isValid(data)) {
+        throw std::invalid_argument("Invalid address data");
```
Comment (vikmeup) [inline src/Zilliqa/Address.cpp]: can we avoid throwing here? per https://github.com/TrustWallet/wallet-core/issues/424

File: src/Zilliqa/AddressChecksum.cpp (hunk 1 above)
Comment (hewigovens) [reply src/Zilliqa/AddressChecksum.cpp]: No, It’s quite different, not an Ethereum address

File: src/Zilliqa/Address.cpp (hunk 2 above)
Comment (hewigovens) [reply src/Zilliqa/Address.cpp]: This method is not exposed to C interface, only used in tests

**Pull request merged**